dbcon/dmlpackageproc/dmlpackageprocessor.cpp

```diff
@@ -287,6 +287,31 @@ int32_t DMLPackageProcessor::tryToRollBackTransaction(uint64_t uniqueId, BRM::Tx
   return weRc;
 }
 
+DMLPackageProcessor::DMLResult DMLPackageProcessor::processPackage(dmlpackage::CalpontDMLPackage& cpackage)
+{
+  auto result = processPackageInternal(cpackage);
+  uint32_t tries = 0;
+  // Try to setup connection and process package one more time.
+  while ((result.result == PP_LOST_CONNECTION) && (tries < 5))
```

Comment:
Is it possible to use do-while here? What do you think @denis0x0D ?

Reply:
I do not think the "do while" is good choice here, because we have to try new connection only if we got "PP_LOST_CONNECTION" error.